nucliadb_vectors/src/labels/mod.rs

```
@@ -0,0 +1,190 @@
+            records_file: RwLock::new(records_file),
+            fst_map,
+        })
+    }
+
+    pub fn get_label(&self, key: &str) -> VectorR<Option<Label>> {
+        let Some(offset) = self.fst_map.get(key) else {
+            return Ok(None);
+        };
+
+        // we have a match, let's get the record
+        let mut records_file = self.records_file.write().unwrap();
+        let _ = records_file.seek(io::SeekFrom::Start(offset))?;
+
+        Ok(bincode::deserialize_from(&*records_file).map(Some)?)
+    }
+}
+
+/// Builds a Labels index
+///
+/// # Example
+///
+/// ```
+/// use nucliadb_vectors::labels::*;
+///
+/// let labels: Vec<Label> = vec![
+///         Label {
+///             key: "key1".to_owned(),
+///             doc_ids: vec![1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
+///         },
+///         Label {
+///             key: "key2".to_owned(),
+///             doc_ids: vec![1, 2, 3, 4, 5],
+///         },
+///         Label {
+///             key: "key3".to_owned(),
+///             doc_ids: vec![7, 8],
+///         },
+///     ];
+///
+/// build_index("labels.fst", "labels.idx", labels.iter());
+/// ```
+pub fn build_index<'a, I>(fst_file_path: &str, records_file_path: &str, labels: I) -> VectorR<usize>
+where
+    I: Iterator<Item = &'a Label>,
+{
+    // create the record and fst files
+    let mut records_writer = BufWriter::new(File::create(records_file_path)?);
+    let fst_file = File::create(fst_file_path)?;
+    let fst_writer = BufWriter::new(fst_file);
+    let mut fst_builder = MapBuilder::new(fst_writer)?;
+
+    // 1. add Label into the records file
+    // 2. add the key into the fst file, the value is the offset in the record file
+    let mut added = 0;
+
+    for label in labels {
+        let start_position: u64 = records_writer.stream_position()?;
+        serialize_into(&mut records_writer, label)?;
+        fst_builder.insert(label.key.clone(), start_position)?;
+        added += 1;
+    }
+    let _ = records_writer.flush();
```

why swallow the error instead of `records_writer.flush()?;` ?